base/applications/mspaint/dialogs.cpp

```diff
@@ -103,10 +103,16 @@ LRESULT CAttributesDialog::OnInitDialog(UINT nMsg, WPARAM wParam, LPARAM lParam,
     newHeight = imageModel.GetHeight();
 
     CheckDlgButton(IDD_ATTRIBUTESRB3, BST_CHECKED);
-    CheckDlgButton(IDD_ATTRIBUTESRB5, BST_CHECKED);
     SetDlgItemInt(IDD_ATTRIBUTESEDIT1, newWidth, FALSE);
     SetDlgItemInt(IDD_ATTRIBUTESEDIT2, newHeight, FALSE);
 
+    HBITMAP hbm = imageModel.CopyBitmap();
```

Comment:
Why do you need to copy the whole bitmap just to read its contents for determining whether it's B/W ? Isn't it possible to just do it from the original image already present in imageModel?
(I haven't checked how you manipulate the "current" bitmap handle elsewhere in mspaint, but having some access to the `m_hBms[m_currInd]` from ImageModel, via an helper function, may be useful :thinking: ... Could be a cast `operator HBITMAP()` or something else?)

Reply:
https://learn.microsoft.com/en-us/windows/win32/api/wingdi/nf-wingdi-selectobject

> An application cannot select a single bitmap into more than one DC at a time.

The problem is that, `imageModel` is already selecting the bitmap.

Reply:
I mean you can still be able to access the underlying (H)BITMAP that you can analyze, no?

Reply:
Double selection might cause the unexpected results.

Reply:
We cannot access the pixels without selection.

Reply:
Synthesizing to imageModel is an option.

Reply:
Synthesized.